Cannot build for PyPI

When trying to build for PyPI distribution the wheel for macos-14-3.10 we get a libomp dependency error and all subsequent builds are cancelled. Full error copied below. Strangely it seems to be targeting mac 11 which is causing the problem.

https://github.com/lephare-photoz/lephare/actions/runs/10386054440/job/28756497226#step:3:277

```
  + delocate-wheel --require-archs arm64 -w /private/var/folders/hw/1f0gcr8d6kn9ms0_wn0_57qc0000gn/T/cibw-run-zdgil_zp/cp310-macosx_arm64/repaired_wheel -v /private/var/folders/hw/1f0gcr8d6kn9ms0_wn0_57qc0000gn/T/cibw-run-zdgil_zp/cp310-macosx_arm64/built_wheel/lephare-0.1.10-cp310-cp310-macosx_11_0_arm64.whl
  INFO:delocate.delocating:Copying library /opt/homebrew/Cellar/libomp/18.1.8/lib/libomp.dylib to lephare/.dylibs/libomp.dylib
  INFO:delocate.delocating:Modifying install name in lephare/_lephare.cpython-310-darwin.so from /opt/homebrew/opt/libomp/lib/libomp.dylib to @loader_path/.dylibs/libomp.dylib
  Fixing: /private/var/folders/hw/1f0gcr8d6kn9ms0_wn0_57qc0000gn/T/cibw-run-zdgil_zp/cp310-macosx_arm64/built_wheel/lephare-0.1.10-cp310-cp310-macosx_11_0_arm64.whl
  Traceback (most recent call last):
    File "/private/var/folders/hw/1f0gcr8d6kn9ms0_wn0_57qc0000gn/T/cibw-run-zdgil_zp/cp310-macosx_arm64/build/venv/bin/delocate-wheel", line 8, in <module>
      sys.exit(main())
    File "/private/var/folders/hw/1f0gcr8d6kn9ms0_wn0_57qc0000gn/T/cibw-run-zdgil_zp/cp310-macosx_arm64/build/venv/lib/python3.10/site-packages/delocate/cmd/delocate_wheel.py", line 110, in main
      copied = delocate_wheel(
    File "/private/var/folders/hw/1f0gcr8d6kn9ms0_wn0_57qc0000gn/T/cibw-run-zdgil_zp/cp310-macosx_arm64/build/venv/lib/python3.10/site-packages/delocate/delocating.py", line 1004, in delocate_wheel
      out_wheel_fixed = _check_and_update_wheel_name(
    File "/private/var/folders/hw/1f0gcr8d6kn9ms0_wn0_57qc0000gn/T/cibw-run-zdgil_zp/cp310-macosx_arm64/build/venv/lib/python3.10/site-packages/delocate/delocating.py", line 839, in _check_and_update_wheel_name
      raise DelocationError(
  delocate.libsana.DelocationError: Library dependencies do not satisfy target MacOS version 11.0:
  /private/var/folders/hw/1f0gcr8d6kn9ms0_wn0_57qc0000gn/T/tmpd38feiv3/wheel/lephare/.dylibs/libomp.dylib has a minimum target of 14.0
                                                              ✕ 1.45s
Error: Command delocate-wheel --require-archs arm64 -w /private/var/folders/hw/1f0gcr8d6kn9ms0_wn0_57qc0000gn/T/cibw-run-zdgil_zp/cp310-macosx_arm64/repaired_wheel -v /private/var/folders/hw/1f0gcr8d6kn9ms0_wn0_57qc0000gn/T/cibw-run-zdgil_zp/cp310-macosx_arm64/built_wheel/lephare-0.1.10-cp310-cp310-macosx_11_0_arm64.whl failed with code 1. 

Error: Process completed with exit code 1.
```

--- a/.github/workflows/publish-to-pypi.yml
+++ b/.github/workflows/publish-to-pypi.yml
@@ -93,7 +93,7 @@ jobs:
         env:
           # CIBW_ARCHS_MACOS: auto universal2
           CIBW_BEFORE_ALL_MACOS: brew install llvm libomp
-          CIBW_ENVIRONMENT_MACOS: PYTHON_VERSION_REQUIRED=${{ matrix.python_version}} MACOSX_DEPLOYMENT_TARGET="10.15" PATH="$(brew --prefix llvm)/bin:$PATH" CFLAGS="-I /usr/local/include -I/usr/local/opt/llvm/include" CXXFLAGS="-I /usr/local/include -I/usr/local/opt/llvm/include -I/opt/homebrew/opt/libomp/include -I/opt/homebrew/opt/llvm/include -I/usr/local/opt/libomp/include" LDFLAGS="-L/usr/local/opt/libomp/lib -L/opt/homebrew/opt/libomp/lib -L /usr/local/lib -L/usr/local/opt/llvm/lib"
+          CIBW_ENVIRONMENT_MACOS: PYTHON_VERSION_REQUIRED=${{ matrix.python_version}} MACOSX_DEPLOYMENT_TARGET="14.0" PATH="$(brew --prefix llvm)/bin:$PATH" CFLAGS="-I /usr/local/include -I/usr/local/opt/llvm/include" CXXFLAGS="-I /usr/local/include -I/usr/local/opt/llvm/include -I/opt/homebrew/opt/libomp/include -I/opt/homebrew/opt/llvm/include -I/usr/local/opt/libomp/include" LDFLAGS="-L/usr/local/opt/libomp/lib -L/opt/homebrew/opt/libomp/lib -L /usr/local/lib -L/usr/local/opt/llvm/lib"
           CIBW_ENVIRONMENT_LINUX: PYTHON_VERSION_REQUIRED=${{ matrix.python_version}}
           CIBW_BUILD: cp${{ matrix.python }}-${{ matrix.platform_id }}
           CIBW_ARCHS: all

--- a/.github/workflows/testing-mac-only.yml
+++ b/.github/workflows/testing-mac-only.yml
@@ -45,7 +45,7 @@ jobs:
     - name: Install lephare and dependencies
       run: |
 
-        export MACOSX_DEPLOYMENT_TARGET="10.15"
+        export MACOSX_DEPLOYMENT_TARGET="14.0"
         export PATH="$(brew --prefix llvm)/bin:$PATH"
         export CFLAGS="-I /usr/local/include -I/usr/local/opt/llvm/include"
         export CXXFLAGS="-I /usr/local/include -I/usr/local/opt/llvm/include -I/opt/homebrew/opt/libomp/include -I/opt/homebrew/opt/llvm/include -I/usr/local/opt/libomp/include"

